Sources_v3/StreamChatUI/ChatChannel/ChatMessageListVC.swift

```diff
@@ -158,6 +177,28 @@ open class ChatMessageListVC<ExtraData: ExtraDataTypes>: ViewController,
         didSelectMessageCell(cell)
     }
 
+    // MARK: - Layout optimizations
+
+    public func setLayoutCached(_ value: Bool) {
+        needsLayoutSwitch = true
+        newLayoutCachedValue = value
+        // invalidating layout while collection view is not in idle state, leads to crash
+        if collectionView.isDecelerating || collectionView.isDragging {
+            return
+        }
```

Comment:
Maybe playing around with `invalidateFlowLayoutDelegateMetrics` and `invalidateFlowLayoutAttributes` can help here? We need to invalidate the layout just to use the new content size, we don't need to  recalculate everything.

Reply:
Can't say that I see any difference, but `invalidateFlowLayoutAttributes` almost exactly what we need. Thanks